chalk-ir/src/macros.rs

```diff
@@ -5,41 +5,41 @@ macro_rules! ty {
     (apply $n:tt $($arg:tt)*) => {
         $crate::TyData::Apply(ApplicationTy {
             name: ty_name!($n),
-            substitution: $crate::Substitution::from(&chalk_ir::interner::ChalkIr, vec![$(arg!($arg)),*] as Vec<$crate::Parameter<_>>),
-        }).intern(&chalk_ir::interner::ChalkIr)
+            substitution: $crate::Substitution::from(&chalk_integration::interner::ChalkIr, vec![$(arg!($arg)),*] as Vec<$crate::Parameter<_>>),
```

Comment:
Wait, I'm confused. `chalk-integration` isn't a dependency (dev or otherwise) of `chalk-ir`. How does this build?

Reply:
Oh, these aren't used in `chalk-ir` at all. Only `chalk-solve`. Can you just move this file there too?

Reply:
Done, but would they fit into `chalk-macros` better?

Reply:
No, those macros are really just the logging stuff. Should probably rename it. Or, maybe better yet, remove and adopt some more standard logging system.. :)

Reply:
Yeah, the macros in `chalk-macros` can just be moved to `chalk-engine`. Or they can just be replaced with tracing or whatever. Separate PR though :)